src/plugins/intel_cpu/src/cpu_tensor.cpp

```diff
@@ -0,0 +1,101 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <memory>
+#include <string>
+#include <map>
+
+#include "cpu_tensor.h"
+#include "ie_blob.h"
+#include "ie_ngraph_utils.hpp"
+#include "openvino/runtime/iremote_tensor.hpp"
+#include "openvino/runtime/properties.hpp"
+
+namespace ov {
+namespace intel_cpu {
+
+//For the beginning probably one contructor should be enough
+Tensor::Tensor(MemoryPtr memptr) : m_memptr{memptr}, m_strides{}, m_strides_once{} {
+    OPENVINO_ASSERT(m_memptr != nullptr);
+
+    // only support plain data format ncsp.
+    auto blocked_desc = m_memptr->getDescWithType<BlockedMemoryDesc>();
+    OPENVINO_ASSERT(blocked_desc, "intel_cpu::Tensor only supports blocked memory descriptor.");
+    OPENVINO_ASSERT(blocked_desc->hasLayoutType(LayoutType::ncsp), "intel_cpu::Tensor only supports memory with ncsp layout.");
+
+    m_element_type = InferenceEngine::details::convertPrecision(blocked_desc->getPrecision());
+    m_shape = blocked_desc->getShape().toPartialShape().to_shape();
+}
+
+void Tensor::set_shape(ov::Shape new_shape) { // this is not as simple as it looks
+// if the ov::Shape is defined, then call cloneWithNewDims to the Memory object memory desc
+// if ov::Shape is undefined (dynamic), we probably have to reintialize the internal Memory object
+// using plain memory format
+    auto& desc = m_memptr->getDesc();
+    const auto newdesc = desc.cloneWithNewDims(new_shape, true);
```

Comment:
AllocatedTensor (from src/inference/src/dev/make_tensor.cpp) which implments ITensor indicates that the implementation defines the expected behavior of set_shape. There it simply switches to a new shape without caring about the old one.
CPUTensor follows it.
As to  "thus we have to create a new memory desc from scratch", can you explain a bit why? What problem with a cloned one instead?

Reply:
> As to "thus we have to create a new memory desc from scratch", can you explain a bit why? What problem with a cloned one instead?

Actually I didn't recognize the `relaxedCheck` flag behavior. Initially we designed this method so that it doesn't allow to clone with new dims if they are not compatible with the existing shape. Ok then, let's leave it as it is.
> 
> btw, ov::Shape could not be dynamic shape. It is always of static shape semantically, though some dim of it could be zero.

Good observation. Let's throw in the case of attempt to get shape from a dynamic shape memory for the beginning. We can always relax this condition later.